### PR TITLE
Update README, RedisJSON. Review migration notes.

### PR DESCRIPTION
**Pull request by banker**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-[![Release](https://img.shields.io/github/release/redis/jedis.svg?sort=semver)](https://github.com/redis/jedis/releases/latest)
+# Jedis
+---
+[![Release](https://img.shields.io/github/release/redis/jedis.svg)](https://github.com/redis/jedis/releases/latest)
```
Comment (sazzad16) [inline README.md:3]:
```suggestion
[![Release](https://img.shields.io/github/release/redis/jedis.svg?sort=semver)](https://github.com/redis/jedis/releases/latest)
```

File: README.md [hunk 2]
```diff
@@ -1,4 +1,6 @@
-[![Release](https://img.shields.io/github/release/redis/jedis.svg?sort=semver)](https://github.com/redis/jedis/releases/latest)
+# Jedis
+---
```
Comment (sazzad16) [inline README.md:2]: Both `#` and `---` draws a line of their own. It becomes two back-to-back lines.
```suggestion
# Jedis

```

File: README.md [hunk 3]
```diff
@@ -7,142 +9,92 @@
...
 
-```xml
-  <repositories>
-    <repository>
-      <id>snapshots-repo</id>
-      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
-    </repository>
-  </repositories>
+```java
+JedisPool pool = new JedisPool("localhost", 6379);
```
Comment (sazzad16) [inline README.md:31]: I'd strongly suggest to use JedisPooled.

File: README.md [hunk 4]
```diff
@@ -7,142 +9,92 @@
...
-
-### Official Releases
+To get started with Jedis, first add it as a dependency in your Java project. If you're using Maven, that looks like this:
 
 ```xml
 <dependency>
     <groupId>redis.clients</groupId>
     <artifactId>jedis</artifactId>
-    <version>3.7.0</version>
+    <version>4.0.0<version>
```
Comment (sazzad16) [inline README.md:24]:
```suggestion
    <version>4.0.0</version>
```

File: README.md [hunk 5]
```diff
@@ -7,142 +9,92 @@
...
-      <groupId>redis.clients</groupId>
-      <artifactId>jedis</artifactId>
-      <version>3.8.0-SNAPSHOT</version>
-    </dependency>
-  </dependencies>
+## Using RedisJSON and Redis modules
+
+Jedis provides support for some of the [Redis modules](https://redis.io/modules), most notably [RedisJSON](https://oss.redis.com/redisjson/) and [RediSearch](https://oss.redis.com/redisearch/).
+
+See the [RedisJSON Jedis Quick Start](docs/redisjson.md) for details.
```
Comment (sazzad16) [inline README.md:51]: We should put JedisPool(ed) and JedisCluster creation together.
We can put these just after that.

File: README.md [hunk 6]
```diff
@@ -7,142 +9,92 @@
...
+Jedis lets you connect to Redis Clusters, supporting the [Redis Cluster Spec](http://redis.io/topics/cluster-spec). To do this, you'll need to connect using `JedisCluster`. See the example below:
+
+```java
+Set<HostAndPort> jedisClusterNodes = new HashSet<HostAndPort>();
+jedisClusterNodes.add(new HostAndPort("127.0.0.1", 7379));
+jedisClusterNodes.add(new HostAndPort("127.0.0.1", 8379));
+JedisCluster jedis = new JedisCluster(jedisClusterNodes);
 ```
 
+Now you can use the `JedisCluster` instance and send commands like you would with a standard pooled connection:
```
Comment (sazzad16) [inline README.md:64]: > like you would with a standard pooled connection

For this, we need JedisPooled earlier.

File: docs/redisjson.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+Let's see how this works.
+
+## Connecting with UnifiedJedis
+
+First, let's create a `UnifiedJedis` instance:
+
+```java
+  HostAndPort config = new HostAndPort(Protocol.DEFAULT_HOST, 6479);
+  PooledJedisConnectionProvider provider = new PooledJedisConnectionProvider(config);
+  UnifiedJedis client = new UnifiedJedis(provider);
```
Comment (sazzad16) [inline docs/redisjson.md:19]:
```suggestion
  JedisPooled client = new JedisPooled(Protocol.DEFAULT_HOST, 6479);
```

File: README.md [hunk 8]
```diff
@@ -7,142 +9,92 @@
...
+You can also check the [latest Jedis Javadocs](https://www.javadoc.io/doc/redis.clients/jedis/latest/index.html).
 
-And you are done!
+## Troubleshooting
 
-## Jedis Cluster
+If you run into trouble or have any questions, we're here to help!
 
-Redis cluster [specification](http://redis.io/topics/cluster-spec) is implemented
+Hit us up on the [Redis Discord Server](http://discord.gg/redis) or [open an issue on GitHub](https://github.com/redis/jedis)
```
Comment (sazzad16) [inline README.md:80]: We are also using GitHub Discussions.

File: docs/3to4-zset-list.md [hunk 9]
```diff
@@ -1,4 +1,4 @@
-### Following SortedSet methods now return Java `List` instead of `Set`:
+### The following sorted set methods now return a Java `List` instead of a `Set`:
```
Comment (sazzad16) [inline docs/3to4-zset-list.md:1]: What's wrong with SortedSet or Sorted Set?

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -155,29 +107,4 @@ Thanks for helping!
...
-included in all copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
-OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
-NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
-HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
-WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
-OTHER DEALINGS IN THE SOFTWARE.
```
Comment (sazzad16) [inline README.md:183]: @gkorland @chayim Your call.

File: README.md [hunk 11]
```diff
@@ -155,29 +107,4 @@ Thanks for helping!
...
+innovative and intelligent tools for profiling Java and .NET applications.
```
Comment (sazzad16) [inline README.md:110]: @gkorland @chayim Do we have any decision about YourKit?

File: docs/3to4.md [hunk 12]
```diff
@@ -1,170 +1,179 @@
...
+  - `GeoRadiusResponse`
+  - `ScanResult`
+  - `Slowlog`
+  - `StreamConsumersInfo`
+  - `StreamEntry`
+  - `StreamGroupInfo`
+  - `StreamInfo`
+  - `StreamPendingEntry`
+  - `StreamPendingSummary`
+  - `Tuple`
```
Comment (sazzad16) [inline docs/3to4.md:103]: To other reviewers:

Too many code formatting annoys me. Am I the only one here?

PS: @banker This is not a change request.

File: docs/redisjson.md [hunk 13]
```diff
@@ -0,0 +1,98 @@
+# RedisJSON Jedis Quick Start
+
+Jedis supports [RedisJSON](https://oss.redis.com/redisjson/) and [RediSearch](https://oss.redis.com/redisearch/).
```
Comment (yangbodong22011) [inline docs/redisjson.md:3]: If only `RedisJSON` and `RediSearch` are mentioned here, then add another Module and need to modify this, so I suggest deleting this line. In the `RedisJSON` documentation, only the usage of `RedisJSON` is provided.

File: README.md [hunk 14]
```diff
@@ -7,142 +9,92 @@
...
 
-```xml
-  <dependencies>
-    <dependency>
-      <groupId>redis.clients</groupId>
-      <artifactId>jedis</artifactId>
-      <version>3.8.0-SNAPSHOT</version>
-    </dependency>
-  </dependencies>
+## Using RedisJSON and Redis modules
```
Comment (yangbodong22011) [inline README.md:47]:
```suggestion
## Using Redis Modules
```

File: README.md [hunk 15]
```diff
@@ -7,142 +9,92 @@
...
+
+Jedis provides support for some of the [Redis modules](https://redis.io/modules), most notably [RedisJSON](https://oss.redis.com/redisjson/) and [RediSearch](https://oss.redis.com/redisearch/).
+
+See the [RedisJSON Jedis Quick Start](docs/redisjson.md) for details.
```
Comment (yangbodong22011) [inline README.md:51]:
```suggestion
Jedis provides support for some of the Redis modules, see the [doc](https://github.com/redis/jedis/blob/8246c42a60/docs/redismodules.md) for details.
```

File: docs/redisjson.md (hunk 13 above)
Comment (sazzad16) [reply docs/redisjson.md:3]: @yangbodong22011 I don't mind this. At some point in history, we did 

https://github.com/redis/jedis/blob/f4b8df496c698b17dc8a009b704de39de8bcfdc7/README.md#L24-L46

So we can afford listing module names that supported through Jedis.

File: README.md [hunk 16]
```diff
@@ -7,142 +9,92 @@
...
-## Community
-
-Meet us on IRC: ##jedis on freenode.net
-
-Join the mailing-list at [http://groups.google.com/group/jedis_redis](http://groups.google.com/group/jedis_redis)
+## What is Jedis?
 
-## So what can I do with Jedis?
-All of the following redis features are supported:
+Jedis is a Java client for [Redis](http://github.com/antirez/redis "Redis") designed for performance and ease of use.
```
Comment (zeekling) [inline README.md:14]: url should be http://github.com/redis/redis

File: README.md [hunk 17]
```diff
@@ -7,142 +9,92 @@
...
-  </dependencies>
+## Using RedisJSON and Redis modules
+
+Jedis provides support for some of the [Redis modules](https://redis.io/modules), most notably [RedisJSON](https://oss.redis.com/redisjson/) and [RediSearch](https://oss.redis.com/redisearch/).
+
+See the [RedisJSON Jedis Quick Start](docs/redisjson.md) for details.
+
+## Connecting to a Redis cluster
+
+Jedis lets you connect to Redis Clusters, supporting the [Redis Cluster Spec](http://redis.io/topics/cluster-spec). To do this, you'll need to connect using `JedisCluster`. See the example below:
```
Comment (zeekling) [inline README.md:55]: `[Redis Cluster Spec](http://redis.io/topics/cluster-spec)` should be `[Redis Cluster Specification](http://redis.io/topics/cluster-spec)`

File: docs/redisjson.md (hunk 13 above)
Comment (sazzad16) [reply docs/redisjson.md:3]: @yangbodong22011 I've re-read and I believe we misunderstood. This doc describes RedisJSON with RediSearch. From that perspective, there's no problem with the statement.

Comment (yangbodong22011) [reply docs/redisjson.md:3]: @sazzad16 yes, agree with you.

**Comment by sazzad16** [comment]
We have waited enough. I'm merging this now.

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**